applications/connectivity_bridge/src/disk/thingy91_cdc_acm.inf.c

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#define MODULE file_inf
+#include "fs_event.h"
+
+#if CONFIG_FS_FATFS_LFN
+#define FILE_NAME         "thingy91_cdc_acm.inf"
+#else
+#define FILE_NAME         "THINGY91.INF"
+#endif
+#define FILE_CONTENTS     file_contents
+#define FILE_CONTENTS_LEN sizeof(file_contents)
+
+static const char file_contents[2759] = {
+	0x3B, 0x20, 0x43, 0x6F, 0x70, 0x79, 0x72, 0x69, 0x67, 0x68,
```

Comment:
You might be interested now, or in the future, to use our infrastructure for embedding files into FW.

Have a look at

ncs/zephyr/tests/application_development/gen_inc_file

It generates a file like

```
0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17,
0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f,
0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27,

```

from a .bin file in that sample's case. But it works for ascii as well I think, although I haven't found any such use.

```
# 1.3 generate_inc_*

# These functions are useful if there is a need to generate a file
# that can be included into the application at build time. The file
# can also be compressed automatically when embedding it.
#
# See tests/application_development/gen_inc_file for an example of
# usage.
function(generate_inc_file
    source_file    # The source file to be converted to hex
    generated_file # The generated file
    )
  add_custom_command(
    OUTPUT ${generated_file}
    COMMAND
    ${PYTHON_EXECUTABLE}
    ${ZEPHYR_BASE}/scripts/file2hex.py
    ${ARGN} # Extra arguments are passed to file2hex.py
    --file ${source_file}
    > ${generated_file} # Does pipe redirection work on Windows?
    DEPENDS ${source_file}
    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
    )
endfunction()

```

we also use it to include certificates into FW.

Reply:
Thanks for the tip! Using this framework instead now, and adding the original files instead of having magic arrays for the driver .cat and .inf files.